packages/tools/src/utilities/annotationFrameRange.ts

```diff
@@ -0,0 +1,84 @@
+import { triggerEvent, eventTarget } from '@cornerstonejs/core';
+import Events from '../enums/Events';
+import { Annotation } from '../types';
+
+/**
+ * This class handles the annotation frame range values for multiframes.
+ * Mostly used for the Video viewport, it allows references to
+ * a range of frame values.
+ *
+ */
+
+export type FramesRange = [number, number] | number;
+
+export default class AnnotationFrameRange {
+  public static frameRangeExtractor = /(\/frames\/|[&?]frameNumber=)([^/&?]*)/i;
```

Comment:
I prefer functional programming over the class here since it does not add any value. But it's fine, just for the future.